packages/@cdktf/hcl2cdk/lib/expressions.ts

```diff
@@ -82,46 +82,48 @@ export async function extractReferencesFromExpression(
       return carry;
     }
 
-    if (scopedIds.includes(corespondingNodeId)) {
+    if (scopedIds.includes(correspondingNodeId)) {
       logger.debug(`skipping '${value}' since it's a scoped variable`);
       return carry;
     }
-    logger.debug(`Found node id '${corespondingNodeId}'`);
-
-    const spotParts = value.split(".");
-    let isThereANumericAccessor = false;
-    const referenceSpotParts = spotParts.filter((part) => {
-      if (!Number.isNaN(parseInt(part, 10))) {
-        isThereANumericAccessor = true;
-        return false;
-      }
+    logger.debug(`Found node id '${correspondingNodeId}'`);
 
-      return !isThereANumericAccessor;
-    });
-    const fullReference = isThereANumericAccessor
-      ? referenceSpotParts.slice(0, 2).join(".")
-      : value;
+    const fullReference = value;
 
     const isVariable = value.startsWith("var.");
-    const useFqn =
-      // Can not use FQN on vars
-      !isVariable &&
-      // Can not use FQN on locals
-      !value.startsWith("local.") &&
-      // If the following character is
-      (input.substr(endPosition + 1, 1) === "*" || // a * (splat) we need to use the FQN
-        input.substr(endPosition, 1) === "[" || // a property access
-        isThereANumericAccessor || // a numeric access
-        fullReference.split(".").length < 3);
+    const canUseFqn =
```

Comment:
I'm okay with this change, but I'm not sure if it's the right way forward. There could still be brittle / broken use-cases we just haven't uncovered. 

I would suggest we at least entertain the idea of using the parseExpression function in hcl2cdkto parse these possible references and using the resulting HCL-AST to answer the questions we have about the reference here :)

Reply:
Ah, that's an interesting idea. I'll give it a shot.